packages/api/src/command/medical/invalid-links/create-invalid-links.ts

```diff
@@ -0,0 +1,37 @@
+import { executeOnDBTx } from "../../../models/transaction-wrapper";
+import { InvalidLinksData } from "../../../domain/invalid-links";
+import { InvalidLinks } from "../../../domain/invalid-links";
+import { InvalidLinksCreate } from "../../../domain/invalid-links";
+import { InvalidLinksModel } from "../../../models/invalid-links";
+import { getInvalidLinks } from "./get-invalid-links";
+import { updateInvalidLinksWithinDBTx } from "./update-invalid-links";
+
+export async function createOrUpdateInvalidLinks({
+  id,
+  cxId,
+  invalidLinks,
+}: {
+  id: string;
+  cxId: string;
+  invalidLinks: InvalidLinksData;
+}): Promise<InvalidLinks> {
+  const invalidLinksCreate: InvalidLinksCreate = {
+    id,
+    cxId,
+    data: invalidLinks,
+  };
+
+  const updateResult = await executeOnDBTx(InvalidLinksModel.prototype, async transaction => {
+    const existingInvalidLinks = await getInvalidLinks({
+      id,
+      cxId,
+      transaction,
+      lock: true,
+    });
+    if (!existingInvalidLinks) return undefined;
+    return updateInvalidLinksWithinDBTx(invalidLinksCreate, existingInvalidLinks, transaction);
+  });
+  if (updateResult) return updateResult.dataValues;
+
+  return await InvalidLinksModel.create(invalidLinksCreate);
```

Comment:
Let's return `.dataValues` when we're returning the domain interface and not the Model, please